code/game/objects/items/devices/scanners.dm

```diff
@@ -65,14 +65,26 @@ GENE SCANNER
 	if(!ismob(viewer) || !viewer.client)
 		return
 	var/list/t_ray_images = list()
-	for(var/obj/O in orange(distance, viewer) )
-		if(HAS_TRAIT(O, TRAIT_T_RAY_VISIBLE))
-			var/image/I = new(loc = get_turf(O))
-			var/mutable_appearance/MA = new(O)
-			MA.alpha = 128
-			MA.dir = O.dir
-			I.appearance = MA
-			t_ray_images += I
+	for(var/turf/turf_to_scan as anything in RANGE_TURFS(distance, viewer))
+		var/list/list_to_scan = turf_to_scan.nullspaced_contents
+
+		for(var/obj/object_to_scan as anything in list_to_scan)
+			if(!HAS_TRAIT(object_to_scan, TRAIT_T_RAY_VISIBLE))
+				continue
+
+			var/mutable_appearance/new_image = new(object_to_scan)
+			new_image.loc = object_to_scan.associated_loc
+			new_image.alpha = 128
+			new_image.dir = object_to_scan.dir
+			t_ray_images += new_image
+
+	for(var/obj/O in orange(distance, viewer))
+		if(HAS_TRAIT(O, TRAIT_T_RAY_VISIBLE) && !(O.associated_loc == O.loc))
+			var/mutable_appearance/new_image = new(O)
+			new_image.loc = get_turf(O)
+			new_image.alpha = 128
+			new_image.dir = O.dir
+			t_ray_images += new_image
```

Comment:
I kinda thought you were using a plane for this, so we could ditch the weird pulsing behavior of trays. has that changed?

Reply:
i tried to and dropped the idea since i wasnt sure of an ok way to do it

Reply:
what was the idea exactly?

Reply:
putting an overlay on everything visible on a t ray plane and making the plane only visible to people with the scanners